src/Core/Types.Filters/Expressions/ComparableOperationHandler.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Linq.Expressions;
+using HotChocolate.Language;
+using HotChocolate.Utilities;
+
+namespace HotChocolate.Types.Filters.Expressions
+{
+    public class ComparableOperationHandler
+        : IExpressionOperationHandler
+    {
+        public bool TryHandle(
+            FilterOperation operation,
+            IInputType type,
+            IValueNode value,
+            Expression instance,
+            ITypeConversion converter,
+            out Expression expression)
+        {
+            if (operation.Type == typeof(IComparable)
+                && value is IValueNode<IComparable> s)
+            {
+                MemberExpression property =
+                    Expression.Property(instance, operation.Property);
+                var parsedValue = type.ParseLiteral(value);
+
+                if (operation.Property.PropertyType
```

Comment:
@PascalSenn I added this converse here since the scalar value might not match the property type.